libcudacxx/include/cuda/std/__type_traits/type_set.h

```diff
@@ -83,6 +83,10 @@ using __type_set_insert = decltype(__set::__bulk_insert::__call<_Ts...>(static_c
 template <class... _Ts>
 using __make_type_set = __type_set_insert<__type_set<>, _Ts...>;
 
+template <class _Ty, class... _Ts>
+_LIBCUDACXX_INLINE_VAR constexpr bool __is_contained_in =
+  _CCCL_TRAIT(_CUDA_VSTD::is_base_of, __type_identity<_Ty>, _CUDA_VSTD::__make_type_set<_Ts...>);
```

Comment:
The new one uses `is_base_of` while the old one was `is_same_v`, do we plan to support inheriting from properties?

Reply:
Oh that is just how @ericniebler implemented the matching for `__type_set` The old one did plain type checking.

I am wondering whether its more efficient the old way but I guess it doesnt really matter